src/lib/zodSchema/signUpSchema.ts

```diff
@@ -0,0 +1,31 @@
+import { z } from 'zod';
+
+export type SignUpSchemaType = typeof SignUpSchema;
+
+const SignUpSchema = z
+  .object({
+    username: z
```

Comment:
The User schema does not contain the username, is the username really necessary in this application?

Reply:
There are other fields that might be needed at signup like the user phone number, address.

Reply:
Yes, these are already underway.

I only created this generic schema based on the fact that you were still working on the mongoose one